src/scripts/OSFramework/Event/WindowOnMessage.ts

```diff
@@ -0,0 +1,26 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace OSFramework.Event {
+	/**
+	 * Class that represents the Post Message on the Window.
+	 *
+	 * @export
+	 * @class WindowMessage
+	 * @extends {Event.AbstractEvent<string>}
+	 */
+	export class WindowMessage extends Event.AbstractEvent<string> {
+		constructor() {
+			super();
+			if (this._windowIsInsideIframe()) {
+				window.addEventListener(GlobalEnum.HTMLEvent.Message, this._windowTrigger.bind(this), true);
+			}
+		}
+
+		private _windowIsInsideIframe(): boolean {
+			return window.self !== window.top;
+		}
```

Comment:
I would remove this a move the condition directly to the `if`.

Reply:
Moved